### ISSUE TITLE
[14.0] auth_saml : cert and key mandatory even if no signature needed

### ISSUE THREAD

**Issue by remi-filament** [OCA/server-auth#315]
## Module
auth_saml

## Describe the bug
When adding a SAML provider, in case you do not want your requests to IdP signed, you do not need cert and private key.
However, if you do not generate these, you will not be able to generate metadata (you get a 500 Internal Server Error).
Also, Signature Algorithm is mandatory and should not be if Sign checkbox is not ticked.

## To Reproduce
**Affected versions**: 14.0

1. Create new SAML provider
2. Fill in IdP metadata
3. Save
4. Click on the link generated for Odoo metadata (in Metadata URL) --> you get a 500 Internal Server Error)

**Expected behavior**
You should be able to generate metadata without setting certificate / private key in case you do not need to sign requests.
In any case, you may want to add some validation checks (which today raise error on logs when trying to access metadata but are not visible to Odoo user) :
- check that both private key and certificate are provided in case you tick Sign checkbox
- validate XML for IdP metadata

Eventually, xmlsec1 is required by pysaml2 (this information is present in readme/INSTALL.rst but not in generated README.rst, which may need to be re-generated).

**Additional context**
```
odootest | 2021-12-13 10:26:30,792 1 ERROR filament10_test werkzeug: Error on request:
odootest | Traceback (most recent call last):
odootest |   File "/opt/odoo/odoo/odoo/addons/base/models/ir_http.py", line 237, in _dispatch
odootest |     result = request.dispatch()
odootest |   File "/opt/odoo/odoo/odoo/http.py", line 807, in dispatch
odootest |     r = self._call_function(**self.params)
odootest |   File "/opt/odoo/odoo/odoo/http.py", line 360, in _call_function
odootest |     return checked_call(self.db, *args, **kwargs)
odootest |   File "/opt/odoo/odoo/odoo/service/model.py", line 94, in wrapper
odootest |     return f(dbname, *args, **kwargs)
odootest |   File "/opt/odoo/odoo/odoo/http.py", line 348, in checked_call
odootest |     result = self.endpoint(*a, **kw)
odootest |   File "/opt/odoo/odoo/odoo/http.py", line 913, in __call__
odootest |     return self.method(*args, **kw)
odootest |   File "/opt/odoo/odoo/odoo/http.py", line 532, in response_wrap
odootest |     response = f(*args, **kw)
odootest |   File "/opt/odoo/additional_addons/auth_saml/controllers/main.py", line 231, in saml_metadata
odootest |     client._metadata_string(
odootest |   File "/opt/odoo/additional_addons/auth_saml/models/auth_saml_provider.py", line 293, in _metadata_string
odootest |     sp_config = self._get_config_for_provider(base_url)
odootest |   File "/opt/odoo/additional_addons/auth_saml/models/auth_saml_provider.py", line 185, in _get_config_for_provider
odootest |     "cert_file": self._get_cert_key_path("sp_pem_public"),
odootest |   File "/opt/odoo/additional_addons/auth_saml/models/auth_saml_provider.py", line 146, in _get_cert_key_path
odootest |     keys_path = model_attachment._full_path(keys.store_fname)
odootest |   File "/opt/odoo/odoo/odoo/addons/base/models/ir_attachment.py", line 77, in _full_path
odootest |     path = re.sub('[.]', '', path)
odootest |   File "/usr/local/lib/python3.9/re.py", line 210, in sub
odootest |     return _compile(pattern, flags).sub(repl, string, count)
odootest | Exception
```

**Comment by theangryangel** [comment]
I'll do my best to sort this out on the Christmas break. Thank you for the bug report.

**Comment by oussjarrousse** [comment]
This is affecting me as well. Thanks @remi-filament for opening the issue. I was wondering what was going on as I did not have access to the python error output!

**Comment by oussjarrousse** [comment]
So I have generated a certificate and a key and uploaded them:

When I leave the "sign" box unchecked (while the fields for certificate and key are still set). the SAML process succeeds.
When i check the "sign" box with the fields for the certificate and key set and hashing set, the SAML process fails with:
"502 Bad Gateway" from my nginx (reverse proxy) after redirecting to: https://odoo.simulalab.org/auth_saml/get_auth_request?pid=1 (pid=1 is Google workspace custom app) but there is no error message in the odoo logs, nor debug messages related to the issue.

**Comment by theangryangel** [comment]
FYI, this is still on my backlog of OCA work at the moment.

**Comment by remi-filament** [comment]
Fixed with #342 on v15.0, with #370 on v14.0
Thank you @vincent-hatakeyama for the rework and @gurneyalex for the backport !
